### DISCUSSION TITLE
Structural Stability Issues in Five-Element Alloy Surface Simulations

### DISCUSSION THREAD

**Discussion by ENGFOX-R** [deepmodeling/deepmd-kit#4873] [Q&A]
Dear experts,  
I'm encountering some challenges while using deepmd-kit for five-element alloy simulations and would greatly appreciate your insights and guidance.

Background:  
Successfully trained a Deep Learning Potential (DLP) containing Al, Nb, Ti, V, and Zr elements.
Performing molecular dynamics simulations using LAMMPS.
Simulation protocol: Structure minimization → NPT ensemble simulation

Observations:
Bulk alloy simulations perform well:
Maintains high proportion of BCC structure during both minimize and NPT stages
Shows good structural stability
Surface simulations show problematic behavior:
[npt_simulation.txt](https://github.com/user-attachments/files/21661083/npt_simulation.txt)

During minimize stage: Structure undergoes BCC → FCC → BCC transformation(figure 1)
During NPT stage: Partial surface atoms scatter into the vacuum layer(figure 2)

Questions:
1.Could this surface atom scattering phenomenon be related to insufficient DLP training data or parameter settings?
2.Are there special treatments or parameter adjustments needed for surface simulations?
3.Are there recommended best practices for surface simulations that I could reference?

Thank you for your time and suggestions. I look forward to your responses.

my training dataset:
![data](https://github.com/user-attachments/assets/b2daa73c-105e-4eed-b3d8-ca42eae9f764)

![mini](https://github.com/user-attachments/assets/080a63bb-11bf-4e32-8715-2aa6f6660862)
figure 1:Structure undergoes BCC → FCC → BCC transformation

![NPT](https://github.com/user-attachments/assets/dc9dbb30-ff70-43b2-984e-bdeb8de40be3)
figure 2:Partial surface atoms scatter into the vacuum layer

**Comment by wanghan-iapcm** [comment]
Have you sampled surface configurations in your training data? do they effectively cover relevant configurations in the surface npt md simulations?

**Reply by ENGFOX-R** [reply]
My training data contains (110) and (111) surface configurations executing variable-cell molecular dynamics (vc-md) at 300K and 600K, whereas LAMMPS performs NPT ensemble simulations at 300K.

**Reply by wanghan-iapcm** [reply]
I would guess you performed AIMD simulation to sampling the training data, which is usually insufficient (not 100% sure because you have not reported how did you sample along the AIMD trajs). I also notice that the total number of training data is limited. 

I would guess the training data are not suffice to cover all relevant configurations in your downstream lammps npt simulations. you may check the accuracy of the model by model deviation https://docs.deepmodeling.com/projects/deepmd/en/master/test/model-deviation.html  and try to improve the model's quality by concurrent learning methods like the dpgen https://github.com/deepmodeling/dpgen

**Comment by ENGFOX-R** [comment]
Thank you very much for your reply, Dr. Wang. I will try to follow your suggestion.

**Comment by 1782552114** [comment]
I am also training the potential function for alloys. May I see the code of your input.json? The potential function results from my code are not ideal.{
    "_comment": " model parameters",
    "model": {
        "type_map": ["Si","Al","Mg"],
        "descriptor": {
            "type": "se_e2_a",
            "sel": "auto",
            "rcut_smth": 0.50,
            "rcut": 6.50,
            "neuron": [64, 128, 256],
            "resnet_dt": false,
            "axis_neuron": 32,
            "seed": 1
        },
        "fitting_net": {
            "neuron": [128,256,512],
            "resnet_dt": true,
            "seed": 1
        }
    },

    "learning_rate": {
        "type": "exp",
        "decay_steps": 5000,
        "start_lr": 0.001,
        "stop_lr": 1e-8
    },

    "loss": {
        "type": "ener",
        "start_pref_e": 0.02,
        "limit_pref_e": 1.0,
        "start_pref_f": 1000,
        "limit_pref_f": 1,
        "start_pref_v": 0.02,
        "limit_pref_v": 1.0
    },

    "training": {
        "training_data": {
            "systems": [],
            "batch_size": "auto",
            "auto_prob": "prob_sys_size"
        },
        "validation_data": {
            "systems": [],
            "batch_size": "auto",
            "numb_btch": 3
        },
        "numb_steps": 500000,
        "seed": 10,
        "disp_file": "lcurve.out",
        "disp_freq": 1000,
        "save_freq": 10000
    }
}
<img width="4769" height="3558" alt="training_loss_comparison" src="https://github.com/user-attachments/assets/375d01db-0d4a-4709-a177-e93f74d19ee7" />
